app/src/main/java/com/dragic/gamehunter/settings/UserSettings.kt

```diff
@@ -0,0 +1,42 @@
+package com.dragic.gamehunter.settings
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import com.dragic.gamehunter.view.theme.AppTheme
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.flow.MutableStateFlow
+import javax.inject.Inject
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+class UserSettings @Inject constructor(
+    @ApplicationContext context: Context
+) {
+
+    val themeStream: MutableStateFlow<AppTheme>
+    var theme: AppTheme by AppThemePreferenceDelegate("app_theme", AppTheme.LIGHT)
+
+    private val preferences: SharedPreferences =
+        context.getSharedPreferences("sample_theme", Context.MODE_PRIVATE)
+
+    init {
+        themeStream = MutableStateFlow(theme)
+    }
+
+    inner class AppThemePreferenceDelegate(
+        private val name: String,
+        private val default: AppTheme,
+    ) : ReadWriteProperty<Any?, AppTheme> {
+
+        override fun getValue(thisRef: Any?, property: KProperty<*>): AppTheme =
+            AppTheme.fromOrdinal(preferences.getInt(name, default.ordinal))
+
+        override fun setValue(thisRef: Any?, property: KProperty<*>, value: AppTheme) {
+            themeStream.value = value
+            preferences.edit {
+                putInt(name, value.ordinal)
+            }
+        }
+    }
```

Comment:
It's nice that you're using delegates, but a simpler solution can be achieved just with regular Kotlin getters and setters: 

```kt
var theme: AppTheme
  get() = TODO()
  set(value) { TODO() }
```

Try implementing it that way.